include/aws/iotdevice/secure_tunneling.h

```diff
@@ -80,6 +86,10 @@ struct aws_secure_tunnel {
     /* The secure tunneling endpoint ELB drops idle connect after 1 minute. We need to send a ping periodically to keep
      * the connection */
     struct aws_task ping_task;
+
+    bool data_callback_flag;
```

Comment:
maybe call this "can_send_data", resetting to true after each send completion.  Also leave a comment noting that it's shared state, or use the pattern where we embed all shared state in a structure called synced_data ala https://github.com/awslabs/aws-c-mqtt/blob/main/include/aws/mqtt/private/client_impl.h#L165